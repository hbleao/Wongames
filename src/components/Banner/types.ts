import { RibbonColors, RibbonSizes } from '../Ribbon/types';

export type ImageProps = {
  src: string;
};

export type BannerProps = {
  image: string;
  title: string;
  subtitle: string;
  buttonLabel: string;
  buttonLink: string;
  ribbon?: string;
  ribbonSize?: RibbonSizes;
  ribbonColor?: RibbonColors;
};
