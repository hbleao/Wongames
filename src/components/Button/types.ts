import React, {
  AnchorHTMLAttributes,
  ButtonHTMLAttributes,
  ReactNode
} from 'react';

type Size = 'small' | 'medium' | 'large';
type ButtonTypes =
  | AnchorHTMLAttributes<HTMLAnchorElement>
  | ButtonHTMLAttributes<HTMLButtonElement>;

export type ButtonProps = {
  size?: Size;
  fullWidth?: boolean;
  icon?: ReactNode;
  as?: React.ElementType;
} & ButtonTypes;

export type WrapperProps = {
  fullWidth?: boolean;
  size?: Size;
  hasIcon: boolean;
};
