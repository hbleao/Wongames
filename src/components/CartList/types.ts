import { GameItemProps } from 'components/GameItem/types';

export type CartListProps = {
  items: GameItemProps[];
  total: string;
};
