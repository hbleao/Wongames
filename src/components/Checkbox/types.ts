import { InputHTMLAttributes } from 'react';

export type CheckboxProps = {
  labelText?: string;
  labelFor?: string;
  isChecked?: boolean;
  labelColor?: 'black' | 'white';
  value?: string | ReadonlyArray<string> | number;
  onCheck?: (status: boolean) => void;
} & InputHTMLAttributes<HTMLInputElement>;

export type LabelProps = {
  labelColor?: 'black' | 'white';
};
