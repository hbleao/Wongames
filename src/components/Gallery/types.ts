export type GalleryImageProps = {
  src: string;
  alt: string;
  label: string;
};

export type GalleryProps = {
  items: GalleryImageProps[];
};

export type ModalProps = {
  isOpen: boolean;
};
