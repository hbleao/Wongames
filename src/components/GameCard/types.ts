import { RibbonColors, RibbonSizes } from 'components/Ribbon/types';

export type GameCardProps = {
  title: string;
  developer: string;
  image: string;
  price: string;
  promotionalPrice?: string;
  favorite?: boolean;
  ribbon?: string;
  ribbonSize?: RibbonSizes;
  ribbonColor?: RibbonColors;
  onFav: () => void;
};

export type PriceProps = {
  isPromotional?: boolean;
};
