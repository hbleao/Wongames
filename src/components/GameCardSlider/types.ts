import { GameCardProps } from 'components/GameCard/types';

export type GameCardSliderProps = {
  items: GameCardProps[];
  color?: 'white' | 'black';
};

export type WrapperProps = {
  color?: 'white' | 'black';
};
