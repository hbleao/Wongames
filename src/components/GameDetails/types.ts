export type Platform = 'mac' | 'linux' | 'windows';

export type Rating = 'BR0' | 'BR10' | 'BR12' | 'BR14' | 'BR16' | 'BR18';

export type GameDetailsProps = {
  platforms: Platform[];
  developer: string;
  releaseDate: string;
  rating: Rating;
  genres: string[];
};
