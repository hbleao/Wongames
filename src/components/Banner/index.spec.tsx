import React from 'react';
import { screen } from '@testing-library/react';

import { Banner } from '.';
import { renderWithTheme } from 'utils/tests/renderWithTheme';
import { RibbonColors, RibbonSizes } from 'components/Ribbon/types';

type MakeSutProps = {
  image?: string;
  title?: string;
  subtitle?: string;
  buttonLabel?: string;
  buttonLink?: string;
  ribbon?: string;
  ribbonSize?: RibbonSizes;
  ribbonColor?: RibbonColors;
};

const makeSut = ({
  image = 'https://source.unsplash.com/user/willianjusten/1042x580',
  title = 'Defy death',
  subtitle = '<p>Play the new <strong>Crashlands</strong> season</p>',
  buttonLabel = 'Buy now',
  buttonLink = '/games/defy-death',
  ribbon,
  ribbonSize,
  ribbonColor
}: MakeSutProps) => {
  const sut = renderWithTheme(
    <Banner
      image={image}
      title={title}
      subtitle={subtitle}
      buttonLabel={buttonLabel}
      buttonLink={buttonLink}
      ribbon={ribbon}
      ribbonSize={ribbonSize}
      ribbonColor={ribbonColor}
    />
  );

  return {
    sut
  };
};

describe('Banner', () => {
  it('should be render a Banner correctly', () => {
    makeSut({});

    const title = screen.getByRole('heading', { name: /defy death/i });
    const subtitle = screen.getByRole('heading', {
      name: /play the new crashlands season/i
    });
    const image = screen.getByRole('img', { name: /defy death/i });

    expect(title).toBeInTheDocument();
    expect(subtitle).toBeInTheDocument();
    expect(image).toBeInTheDocument();
    expect(title).toMatchSnapshot();
  });

  it('should be render a Banner correctly', () => {
    makeSut({
      ribbon: '10% off',
      ribbonSize: 'small',
      ribbonColor: 'secondary'
    });

    const ribbon = screen.getByText(/10% off/i);

    expect(ribbon).toBeInTheDocument();
    expect(ribbon).toHaveStyle({ backgroundColor: '#3CD3C1' });
    expect(ribbon).toHaveStyle({
      height: '2.6rem',
      fontSize: '1.2rem'
    });
  });
});
