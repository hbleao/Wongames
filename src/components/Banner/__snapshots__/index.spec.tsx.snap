// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner should be render a Banner correctly 1`] = `
.c0 {
  font-size: 1.8rem;
  font-weight: 600;
  color: #FAFAFA;
}

@media (max-width:1170px) {

}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (min-width:768px) {
  .c0 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {

}

<h2
  class="c0"
>
  Defy death
</h2>
`;
